pytest_datadir/plugin.py

```diff
@@ -1,38 +1,22 @@
 import os
+import pathlib
 import shutil
 
 import pytest
 
 
-class TestDataDir(object):
-
-    def __init__(self, global_dir, module_dir, tmpdir):
-        self.global_dir = global_dir
-        self.module_dir = module_dir
-        self.tmpdir = tmpdir.strpath
-        assert os.path.isdir(self.module_dir) or os.path.isdir(self.global_dir),\
-            'neither {} or {} are valid data directories'.format(self.global_dir, self.module_dir)
+@pytest.fixture
+def global_datadir(request):
```

Comment:
I see the motivation behind it. Basically if you have many `test_*.py` files using the same test resource file, without the "global" `data` dir you would be forced to copy the same file across many test data directories, which obviously is not a good practice. I vote to keep it.